File: SerenityExamples.Web/Modules/ServerTypes/Tables/OrdersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrdersRow } from "./OrdersRow";

export interface OrdersColumns {
    OrderDate: Column<OrdersRow>;
    OrderNumber: Column<OrdersRow>;
    CustomerNumber: Column<OrdersRow>;
    SalesNumber: Column<OrdersRow>;
    Total: Column<OrdersRow>;
}

export class OrdersColumns extends ColumnsBase<OrdersRow> {
    static readonly columnsKey = 'Tables.Orders';
    static readonly Fields = fieldsProxy<OrdersColumns>();
}

File: SerenityExamples.Web/Modules/Tables/Orders/OrdersColumns.cs
namespace SerenityExamples.Tables.Columns;

[ColumnsScript("Tables.Orders")]
[BasedOnRow(typeof(OrdersRow), CheckNames = true)]
public class OrdersColumns
{
    [EditLink]
    public DateTime OrderDate { get; set; }

    [EditLink]
    public string OrderNumber { get; set; }

    [EditLink]
    public string CustomerNumber { get; set; }

    [EditLink]
    public string SalesNumber { get; set; }

    [EditLink]
    public int Total { get; set; }
}